lib/faraday/adapter/em_synchrony.rb

```diff
@@ -26,6 +26,11 @@ def call(env)
         # Queue requests for parallel execution.
         if env[:parallel_manager]
           env[:parallel_manager].add(request, http_method, request_config(env)) do |resp|
+            if (req = env[:request]).stream_response?
```

Comment:
I still can't find a nice way to make this work.
If the `call` method you sketched above was located in `Faraday::Adapter` then sub-classes would need to have the same signature for `perform_request` which doesn't seem to be the case.
The same applies to `perform_stream_request` which - if I understood correctly - should delegate to `perform_request` unless overridden in a sub-class.
The only way I can currently think of to make this work is to copy the above code snippet into the `Adapter` sub-classes and have individual implementations of `perform_streaming_request` which would have a similar signature as `perform_request`.
But maybe I'm just overlooking something.

Would you be OK with merging even in the current unDRY state of the code?
Maybe someone comes up with a nicer solution later, I'm currently not seeing a proper way out.